x-pack/plugins/apm/server/lib/service_map/get_service_map_from_trace_ids.ts

```diff
@@ -9,41 +9,39 @@ import {
   SERVICE_ENVIRONMENT,
   SERVICE_NAME,
 } from '../../../common/elasticsearch_fieldnames';
-import {
-  Connection,
-  ConnectionNode,
-  ServiceConnectionNode,
-} from '../../../common/service_map';
+import { Connection, ConnectionNode } from '../../../common/service_map';
 import { Setup } from '../helpers/setup_request';
 import { fetchServicePathsFromTraceIds } from './fetch_service_paths_from_trace_ids';
 
-export function getConnections(
-  paths?: ConnectionNode[][],
-  serviceName?: string,
-  environment?: string
-) {
+export function getConnections({
+  paths,
+  serviceName,
+  environment,
+}: {
+  paths: ConnectionNode[][] | undefined;
+  serviceName: string | undefined;
+  environment: string | undefined;
+}) {
   if (!paths) {
     return [];
   }
 
   if (serviceName || environment) {
     paths = paths.filter((path) => {
       return path.some((node) => {
```

Comment:
it would be good to explicitly include external destinations here. I don't think it will change the result but it makes it more easy to understand for the reader.

Reply:
🤪 but haven't you said that we shouldn't do it [here](https://github.com/elastic/kibana/pull/77757#discussion_r496193630)

Reply:
What I mean is that we shouldn't check external destinations at all. What your code was doing was checking for `span.type != 'external'` only when environment is ENVIRONMENT_NOT_DEFINED (iirc). `span.type == 'external'` does not mean anything useful in this context, as far as I know.

Reply:
> it would be good to explicitly ~~include~~ exclude external destinations here

and what did you mean by this?